docs/guides/typescript.md

```diff
@@ -3,18 +3,16 @@
 As XState is written in [TypeScript](https://www.typescriptlang.org/), strongly typing your statecharts is useful and encouraged. Consider this light machine example:
 
 ```typescript
-// The events that the machine handles
-type LightEvent =
-  | { type: 'TIMER' }
-  | { type: 'POWER_OUTAGE' }
-  | { type: 'PED_COUNTDOWN'; duration: number };
-
-// The context (extended state) of the machine
-interface LightContext {
-  elapsed: number;
-}
-
-const lightMachine = createMachine<LightContext, LightEvent>({
+const lightMachine = createMachine({
+  schema: {
+    // The context (extended state) of the machine
+    context: t<{ elapsed: number }>(),
+    // The events this machine handles
+    events: {} as
+      | { type: 'TIMER' }
+      | { type: 'POWER_OUTAGE' }
+      | { type: 'PED_COUNTDOWN'; duration: number }
+  },
```

Comment:
I would probably stick with a single way of providing those types throughout the documentation. Using stuff consistently raises less questions from the readers